torch_xla/distributed/xla_multiprocessing.py

```diff
@@ -387,17 +393,27 @@ def spawn(fn,
     return _run_direct(fn, args, nprocs, join, daemon, start_method)
 
   pf_cfg = _pre_fork_setup(nprocs)
+  result = None
   if pf_cfg.num_devices == 1:
     _start_fn(0, pf_cfg, fn, args)
   else:
-    return torch.multiprocessing.start_processes(
+    result = torch.multiprocessing.start_processes(
         _mp_start_fn,
         args=(pf_cfg, fn, args),
         nprocs=pf_cfg.num_devices,
         join=join,
         daemon=daemon,
         start_method=start_method)
 
+  # For GPU, xenv.WORKERS are set in the launcher and then get carried to the children.
```

Comment:
:grimacing: this sounds like a nightmare to debug. Thanks for tracking it down. I'll make sure we don't make the same sort of mistake in PJRT

Reply:
yea, it is...